cmd/helm-sync/main.go

```diff
@@ -112,13 +154,21 @@ func main() {
 			Namespace:       *flNamespace,
 			DeployNamespace: *flDeployNamespace,
 			Values:          *flValues,
+			ValuesMergeMode: *flValuesMergeMode,
 			IncludeCRDs:     *flIncludeCRDs,
 			Auth:            configsync.AuthType(*flAuth),
 			HydrateRoot:     *flRoot,
 			Dest:            *flDest,
 			UserName:        *flUsername,
 			Password:        *flPassword,
+
+			RefreshCM:   refreshCM,
+			ValuesFrom:  valuesFrom,
+			CMValuesMap: cmValuesMap,
```

Comment:
This is wrapped in a mutex, but then we're still passing in a pointer to the map here. The struct is going to use this pointer without synchronization.

Related to the other comment, I suggest rewriting this to communicate using channels.